### DISCUSSION TITLE
Centre Text

### DISCUSSION THREAD

**Discussion by ghost** [Bodmer/TFT_eSPI#1978] [Q&A - Fonts] [answered]
How can i print a word in the centre of my display?
using the ST7735 library I use  tft.setCursor(tft.width() / 2, 15, true);
that prints any length word in the top centre of my display
but the eSPI library starts the word in the centre

ive tried using the string align example but still cant figure it out

**Comment by zuhairmst** [comment]
Use the function setTextDATUM() with MC_DATUM which means middle centre.

**Comment by ghost** [comment]
How do I print something Top Centre and 15px from top?

**Comment by zuhairmst** [comment]
tft.setTextDatum(TC_DATUM);
tft.setCursor(width/2,15);

**Comment by Bodmer** [comment, marked as answer]
The print stream cannot be used with text formatting for technical reasons. Use drawString():
```
tft.setTextDatum(MC_DATUM);
tft.drawString("Hello", tft.width() / 2, 15);
```

**Comment by ghost** [comment]
thanks guys, fixed my last problem centering a number with:
`   tft.setTextDatum(MC_DATUM);
   tft.drawString(subPage +1, tft.width() / 2, tft.height() / 2);`

**Comment by hitecSmartHome** [comment]
What if the text is wrapping? I need to center the new line too.

Instead of this

```
|    This is a |
| centered te  |
| xt           |
```

I need something like this

```
|   This is a  |
|  centered te |
|      xt      |
```

Any idea?

If i do it like this

```cpp
tft.setTextWrap(true,true);
tft.setCursor(20,90);
tft.print("This is a centered text");
```

It will wrap and the new line begins at the beginning of the display.

I have tried scrolling but the task does not get enough time so the scrolling is lagging sometimes. ( I don't want to increase the priority just for that )
This is not ideal. It would be good if there was hardware scrolling in st7735 :(

**Comment by hitecSmartHome** [comment]
I also have this setup

```
| [ ~ ] texthere |
| [ ~ ] texthere |
|                |
```

The rectangle is an icon. If the first text wraps, it will start inside the icon. How could I prevent that?

**Worst case**

```
| [ ~ ] texthere |
| [wr]apshere    |
| [ ~ ] texthere |
|                |
```

**Good case**
```
| [ ~ ] texthere  |
| [ ~ ] wrapshere |
| [ ~ ] texthere  |
|                 |
```
